jba/src/inspections/analysis.py

```diff
@@ -1,42 +1,59 @@
 from collections import Counter
+from dataclasses import dataclass
 from itertools import islice
-from typing import Set, List, Tuple
+from typing import Set, List, Optional, Dict
 
 import pandas as pd
 
 from core.src.model.column_name import SubmissionColumns
 from core.src.model.quality.issue.hyperstyle_issue import HyperstyleIssue
 from jba.src.models.edu_columns import EduColumnName
+from jba.src.visualization.common import get_edu_name_columns
 
 
-def find_unique_inspections(group: pd.DataFrame) -> Set[str]:
-    return set(
-        group[EduColumnName.INSPECTIONS.value]
-        .apply(lambda inspections: {inspection.code for inspection in inspections})
-        .explode()
-        .unique()
+def find_unique_inspections(group: pd.DataFrame, file: Optional[str]) -> Set[str]:
+    group_inspections = group[EduColumnName.INSPECTIONS.value].apply(
+        lambda inspections: {
+            inspection.code
+            for file_path, file_inspections in inspections.items()
+            for inspection in file_inspections
+            if file is None or file_path == file
+        }
```

Comment:
this lambda is almost the same at least in several places in this file, can we move it into a function?